spec/models/doorkeeper/access_token_spec.rb

```diff
@@ -143,6 +143,12 @@ module NoGenerate
         subject.resource_owner_id = nil
         expect(subject).to be_valid
       end
+
+      it 'is valid without resource_owner_id' do
+        # For resource owner credentials flow
+        subject.application_id = nil
+        expect(subject).to be_valid
+      end
```

Comment:
Updated. Make sure when testing to include the initializer from `spec/dummy/config/initializers/active_record_belongs_to_required_by_default.rb`